src/SelfScreener/GeneralSymptoms.tsx

```diff
@@ -48,51 +54,53 @@ const GeneralSymptoms: FunctionComponent = () => {
     navigation.navigate(SelfScreenerStackScreens.GeneralSymptomsSummary)
   }
 
+  const noSymptomsSelected =
+    [...primarySymptoms, ...secondarySymptoms, ...otherSymptoms].length === 0
   return (
     <ScrollView>
       <GlobalText>
         {t("self_screener.emergency_symptoms.select_any")}
       </GlobalText>
       <SymptomCheckbox
         label={symptomToString(FEVER_OR_CHILLS)}
-        onPress={() => updateGeneralSymptoms(FEVER_OR_CHILLS)}
-        checked={generalSymptoms.includes(FEVER_OR_CHILLS)}
+        onPress={() => updatePrimarySymptoms(FEVER_OR_CHILLS)}
+        checked={primarySymptoms.includes(FEVER_OR_CHILLS)}
       />
       <SymptomCheckbox
         label={symptomToString(DIFFICULTY_BREATHING)}
-        onPress={() => updateGeneralSymptoms(DIFFICULTY_BREATHING)}
-        checked={generalSymptoms.includes(DIFFICULTY_BREATHING)}
+        onPress={() => updatePrimarySymptoms(DIFFICULTY_BREATHING)}
+        checked={primarySymptoms.includes(DIFFICULTY_BREATHING)}
       />
       <SymptomCheckbox
         label={symptomToString(COUGH)}
-        onPress={() => updateGeneralSymptoms(COUGH)}
-        checked={generalSymptoms.includes(COUGH)}
+        onPress={() => updatePrimarySymptoms(COUGH)}
+        checked={primarySymptoms.includes(COUGH)}
       />
       <SymptomCheckbox
         label={symptomToString(LOSS_OF_SMELL_TASTE_APPETITE)}
-        onPress={() => updateGeneralSymptoms(LOSS_OF_SMELL_TASTE_APPETITE)}
-        checked={generalSymptoms.includes(LOSS_OF_SMELL_TASTE_APPETITE)}
+        onPress={() => updateSecondarySymptoms(LOSS_OF_SMELL_TASTE_APPETITE)}
```

Comment:
Wondering if it would be too overkill to expose `one` method from the context to update the symptoms and then make the determination there on where to store it. And the same to determine if a symptom has been selected. I feel like the separation is priceless for the decision in what the symptoms mean but we could make the view layer simpler by hiding that abstraction. It would probably make a pretty gnarly switch but the types are 💯  soooo it might not be that bad.

Reply:
This was actually a fun refactor. I love it!